scarb/src/bin/scarb/commands/test.rs

```diff
@@ -7,5 +7,16 @@ use crate::args::TestArgs;
 
 #[tracing::instrument(skip_all, level = "info")]
 pub fn run(args: TestArgs, config: &Config) -> Result<()> {
-    ops::execute_test_subcommand(&args.args, config)
+    let ws = ops::read_workspace(config.manifest_path(), config)?;
+
+    if let Ok(package) = ws.current_package() {
+        // Command run from package context.
+        ops::execute_test_subcommand(package, &args.args, config, &ws)
+    } else {
+        // Command run from workspace context.
+        ws.members()
+            .map(|package| ops::execute_test_subcommand(&package, &args.args, config, &ws))
```

Comment:
I wonder if we shouldn't print a message saying "running tests for package {package}..." wdyt?